scripts/infra/cloud-instance.sh

```diff
@@ -119,6 +119,22 @@ ec2__terminate() {
         --region "$EC2_REGION"
 }
 
+wait_ssh_listen() {
+    calculate_cloud_public_dns "$1"
+    if ! command -v nc > /dev/null 2>&1; then
+        echo "nc not on the system path, using a simple wait, install nc for something more precise!"
+        echo "Waiting..."; sleep 30
+        return
+    fi
+    while true; do
+        echo "Waiting for ssh..."
+        sleep 5
+        if nc -w 5 "$PUBLIC_DNS" 22; then
+            break
+        fi
+    done
```

Comment:
Why not just run `ssh` like so? `ssh -o ConnectTimeout=5 user@address exit` This way no deps on nc

Reply:
oh sure yeah, only reason I didn't do it like that is ssh includes the handshake time in that, and so you can get a spurious failure when everything is ok. But in the grand scheme of things another 5 seconds doesnt matter that much vs the dep. so Ill just switch it